src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1083 @@
+        this->actor.world.pos.y = npcAction->startPos.y;
+        this->actor.world.pos.z = npcAction->startPos.z;
+
+        rotY = npcAction->rot.y;
+        this->actor.shape.rot.y = rotY;
+        this->actor.world.rot.y = rotY;
+    }
+}
+
+void func_80969DDC(DemoDu* this, AnimationHeader* animation, u8 mode, f32 morphFrames, s32 arg4) {
+    f32 startFrame;
+    s16 lastFrame;
+    f32 endFrame;
+    f32 playSpeed;
+
+    lastFrame = Animation_GetLastFrame(animation);
+    if (arg4 == 0) {
+        startFrame = 0.0f;
+        endFrame = lastFrame;
+        playSpeed = 1.0f;
+    } else {
+        endFrame = 0.0f;
+        playSpeed = -1.0f;
+        startFrame = lastFrame;
+    }
+    Animation_Change(&this->skelAnime, animation, playSpeed, startFrame, endFrame, mode, morphFrames);
+}
+
+void DemoDu_InitCs_FireMedallion(DemoDu* this, GlobalContext* globalCtx) {
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06011CA8, &D_06006EB0, NULL, NULL, 0);
+    this->actor.shape.yOffset = -10000.0f;
+    DemoDu_SetEyeTexIndex(this, 1);
+    DemoDu_SetMouthTexIndex(this, 3);
+}
+
+// A.k.a Warp portal
+void DemoDu_CsFireMedallion_SpawnDoorWarp(DemoDu* this, GlobalContext* globalCtx) {
+    f32 posX = this->actor.world.pos.x;
+    f32 posY = this->actor.world.pos.y;
+    f32 posZ = this->actor.world.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0, 0, 0, 2);
+}
+
+// Gives the Fire Medallion to Link.
+void func_80969F38(DemoDu* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 posX = player->actor.world.pos.x;
+    f32 posY = player->actor.world.pos.y + 80.0f;
+    f32 posZ = player->actor.world.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, posX, posY, posZ, 0, 0, 0,
+                       DEMO_EFFECT_MEDAL_FIRE);
+    Item_Give(globalCtx, ITEM_MEDALLION_FIRE);
+}
+
+void func_80969FB4(DemoDu* this, GlobalContext* globalCtx) {
+    this->actor.shape.yOffset += 250.0f / 3.0f;
+}
+
+// Gives the Fire Medallion to Link too.
+void DemoDu_CsFireMedallion_AdvanceTo01(DemoDu* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    Player* player;
+    s16 rotY;
+
+    if ((gSaveContext.chamberCutsceneNum == 1) && (gSaveContext.sceneSetupIndex < 4)) {
+        player = PLAYER;
+
+        this->updateIndex = CS_FIREMEDALLION_SUBSCENE(1);
+        globalCtx->csCtx.segment = D_8096C1A4;
+        gSaveContext.cutsceneTrigger = 2;
+        Item_Give(globalCtx, ITEM_MEDALLION_FIRE);
+
+        rotY = this->actor.world.rot.y + 0x8000;
+        player->actor.shape.rot.y = rotY;
+        player->actor.world.rot.y = rotY;
+    }
+}
+
+void DemoDu_CsFireMedallion_AdvanceTo02(DemoDu* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* npcAction;
+
+    if (globalCtx->csCtx.state != 0) {
+        npcAction = globalCtx->csCtx.npcActions[2];
+        if (npcAction != NULL && npcAction->action != 1) {
```

Comment:
```suggestion
    if (globalCtx->csCtx.state != 0) {
        CsCmdActorAction* npcAction = globalCtx->csCtx.npcActions[2];
        
        if ((npcAction != NULL) && (npcAction->action != 1)) {
```

And similar for later functions.